lib/vm/src/module.rs

```diff
@@ -6,21 +6,28 @@
 
 use indexmap::IndexMap;
 use loupe::MemoryUsage;
+#[cfg(feature = "enable-rkyv")]
+use rkyv::{
+    de::SharedDeserializer, ser::Serializer, ser::SharedSerializer, Archive, Archived,
+    Deserialize as RkyvDeserialize, Fallible, Serialize as RkyvSerialize,
+};
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 use std::fmt;
 use std::iter::ExactSizeIterator;
 use std::sync::atomic::{AtomicUsize, Ordering::SeqCst};
 use std::sync::Arc;
 use wasmer_types::entity::{EntityRef, PrimaryMap};
+#[cfg(feature = "enable-rkyv")]
+use wasmer_types::ArchivableIndexMap;
 use wasmer_types::{
     CustomSectionIndex, DataIndex, ElemIndex, ExportIndex, ExportType, ExternType, FunctionIndex,
     FunctionType, GlobalIndex, GlobalInit, GlobalType, ImportIndex, ImportType, LocalFunctionIndex,
     LocalGlobalIndex, LocalMemoryIndex, LocalTableIndex, MemoryIndex, MemoryType, SignatureIndex,
     TableIndex, TableInitializer, TableType,
 };
 
-#[derive(Debug, Clone, MemoryUsage)]
+#[derive(Debug, Clone, MemoryUsage, RkyvSerialize, RkyvDeserialize, Archive)]
```

Comment:
We probably want this as a cfg flag:

```suggestion
#[derive(Debug, Clone, MemoryUsage)]
#[cfg_attr(
    feature = "enable-rkyv",
    derive(RkyvSerialize, RkyvDeserialize, Archive)
)]
```